src/Scene/AnimationPlayer.js

```diff
@@ -0,0 +1,204 @@
+};
+
+const frameEvent = new CustomEvent('frameAnimation');
+const stopEvent = new CustomEvent('stopAnimation');
+const endEvent = new CustomEvent('endAnimation');
+
+/**
+ * AnimationPlayer
+ * It can play, pause or stop animation
+ */
+class AnimationPlayer {
+    constructor(dom) {
+        this.dom = dom;
+        this.id = null;
+        this.keyframe = 0;
+        this.animation = null;
+        this.resolve = null;
+        this.promise = new Promise(r => this.resolve = r);
+        this.state = PLAYER_STATE.STOP;
+        this.callBack = null;
+        this.callBackFrame = () => {};
+    }
+
+    /**
+     * Gets the interval time.
+     *
+     * @return {Number}  The interval time is the time between consecutive frames
+     */
+    static get intervalTime() {
+        return INTERVAL_TIME;
+    }
+
+    isPlayed() {
+        return this.state === PLAYER_STATE.PLAY;
+    }
+
+    isStopped() {
+        return this.state === PLAYER_STATE.STOP;
+    }
+
+    isEnded() {
+        return this.state === PLAYER_STATE.END;
+    }
+
+    // Public functions
+
+    /**
+     * { Start animation }
+     *
+     * @param      {Animation} The animation to play
+     * @return     {Promise}  Promise is resolved when animation is stopped or finished
+     */
+    play(animation, callBack) {
+        this.callBack = callBack;
+        this.animation = animation;
+        if (this.animation.animate) this.callBackFrame = () => this.animation.animate(this.keyframe);
+        setPlayerState.bind(this)(PLAYER_STATE.PLAY);
+        resetTimer.bind(this)();
+        this.id = setInterval(this.frame.bind(this), INTERVAL_TIME);
+        this.promise = new Promise(r => this.resolve = r);
+        return this.promise;
+    }
+
+    /**
+     * { Stop current animation }
+     *
+     * @return  {Promise}  Promise is resolved when animation is stopped or finished
+     */
+    stop() {
+        setPlayerState.bind(this)(PLAYER_STATE.STOP);
+        finishAnimation.bind(this)();
+        this.dom.dispatchEvent(stopEvent);
+        return this.promise;
+    }
+
+    /**
+     * { Pause current animation }
+     */
+    pause() {
+        setPlayerState.bind(this)(PLAYER_STATE.PAUSE);
+        clearInterval(this.id);
+        this.id = undefined;
+    }
+
+    /**
+     * { this function is executed with each frame }
+     */
+    frame() {
+        if (this.keyframe < this.animation.duration) {
+            this.callBackFrame();
+            this.keyframe++;
+            this.dom.dispatchEvent(frameEvent);
+        }
+        else {
+            setPlayerState.bind(this)(PLAYER_STATE.END);
+            finishAnimation.bind(this)();
+            this.dom.dispatchEvent(endEvent);
+        }
+    }
+}
+
+/**
+ * { Animation }
+ *
+ * @class      Animation (name)
+ * @param      {Number}  duration  The animation's duration
+ * @param      {Object}  root      The root object to animate
+ * @param      {String}  name      The animation's name
+ */
```

Comment:
10 lines of periphrases do not make a good documentation.

If you add documentation, make it useful:
* explain the purpose of Animation: move objects? ik based animation? lerp between values? generic?
* specify units (duration is in seconds? days? µs ?)
* why|when should I use (not) `Animation`?